File: testing/src/bucket.cpp
/*
class Bucket {
public:
	Bucket();
	Bucket(const std::vector<Block>& blocks);

	std::vector<Block> get_blocks();
	void set_blocks(const std::vector<Block>& blocks);
private:
	std::vector<Block> blocks;
};
*/

#include "bucket.h"

Bucket::Bucket()
{
	for (int i = 0; i < BLOCKS_PER_BUCKET; ++i) {
		blocks.push_back(Block(0, i, std::vector<uint8_t> ()));
	}
}

Bucket::Bucket(const std::vector<Block>& blocks)
{
	this->blocks = blocks;
}

std::vector<Block> Bucket::get_blocks()
{
	return blocks;
}

void Bucket::set_blocks(const std::vector<Block>& blocks)
{
	this->blocks = blocks;
}

File: Java_code/src/com/pathoram/Bucket.java
package com.pathoram;

import java.io.Serializable;
import java.util.ArrayList;

public class Bucket implements Serializable{

	/**
	 * implement Serializable in order to write bucket object to file
	 * bucket contains Z blocks
	 */
	private static final long serialVersionUID = 1L;
	
	private ArrayList<Block> blocks;

	public Bucket(){
		this.blocks = new ArrayList<Block>(Configs.Z);
		for(int i=0;i<Configs.Z;i++){
			this.blocks.add(new Block());
		}
	}
	
	public Bucket(ArrayList<Block> blocks){
		this.blocks = blocks;
	}
	
	public ArrayList<Block> getBlocks() {
		return blocks;
	}

	public void setBlocks(ArrayList<Block> blocks) {
		this.blocks = blocks;
	}

}
